QueenCam/QueenCam/Presentation/Views/Camera/CameraView.swift

```diff
@@ -399,8 +401,23 @@ extension CameraView: View {
         previewModel.startCapture()
       }
     }
+    // 연결 종료 시 가이딩 초기화
+    .onChange(of: connectionViewModel.networkState) { _, newState in
+      guard let newState else { return }
+      if newState == .host(.cancelled)
+        || newState == .viewer(.cancelled)
+        || newState == .host(.lost)
+        || newState == .viewer(.lost)
+        || newState == .host(.stopped)
+        || newState == .viewer(.stopped) {
+        // 가이딩 초기화
+        penViewModel.reset()
+        frameViewModel.deleteAll()
+        referenceViewModel.onDelete()
+      }
+    }
```

Comment:
```suggestion
    // 연결 종료 시 가이딩 초기화
    .onChange(of: connectionViewModel.networkState) { _, newState in
      guard let newState else { return }
      if newState.neededToResetGuiding  {
        // 가이딩 초기화
        penViewModel.reset()
        frameViewModel.deleteAll()
        referenceViewModel.onDelete()
      }
    }
```

```swift
/// CameraView.swift

extension NetworkState {
  var neededToResetGuiding: Bool {
    newState == .host(.cancelled)
        || newState == .viewer(.cancelled)
        || newState == .host(.lost)
        || newState == .viewer(.lost)
        || newState == .host(.stopped)
        || newState == .viewer(.stopped)
  }
}
```

만약에 리팩토링한다면 이런식으로도 가능할 것 같습니다!!